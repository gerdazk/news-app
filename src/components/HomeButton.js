import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";

function HomeButton({ text }) {
  return (
    <Link to="/" className="homeButton" data-testid="defaultButton">
      {text}
    </Link>
  );
}

export default HomeButton;

HomeButton.propTypes = {
  text: PropTypes.string,
};
