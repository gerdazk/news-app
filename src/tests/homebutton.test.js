import HomeButton from "../components/HomeButton";
import React from "react";
import { mount } from "enzyme";
import { BrowserRouter as Router } from "react-router-dom";

describe("Home Button component: ", () => {
  it("To match snapshot", () => {
    const wrapper = mount(
      <Router>
        <HomeButton text="Hello world" />
      </Router>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
