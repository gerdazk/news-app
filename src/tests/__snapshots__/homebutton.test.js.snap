// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Button component:  To match snapshot 1`] = `ReactWrapper {}`;
